### ISSUE TITLE
xija_gui_fit crashes post-shiny on Linux when updating certain plots

### ISSUE THREAD

**Issue by jzuhone** [sot/xija#105]
After the shiny upgrade, `xija_gui_fit` can crash without any error messages on Linux. 

Steps to reproduce:

Run `xija_gui_fit` on the 1DPAMZT model like so:

```
xija_gui_fit --days=365 --stop=2021:018:00:00:00 dpa_model_spec.json
```

Hit the `Histogram` button, and then check the boxes on and off to mask radzones, FMT1, or hit the `Redraw` button. If you do this a few times in close succession, eventually the whole GUI will crash. A similar thing happens if you click and un-click the radzone draw checkbox on the main GUI. 

I've reproduced this both running `xija_gui_fit` on the HEAD LAN and also on a local machine without SSH. @Gregg140 also saw this. 

This potentially has something to do with the button clicking trying to redraw the plot before a previous redraw operation has completed. Oddly, this does not occur on macOS. 

This could potentially be solved by adding threads for the plotting, but I am certainly not an expert on that and the code searches I have done for this seem pretty complex. 

Not clear what changed post-shiny to cause this.

**Comment by taldcroft** [comment]
@jzuhone - I'm not completely clear on what you mean by "changed post-shiny". Do you mean what happened with the big Py 3.8 2020.14 (https://github.com/sot/skare3/pull/545) promotion, or what happened between 2020.14 and now (namely the 2021.1 release)? 

The matplotlib: 2.0.2 -> 3.2.2 update in 2020.14 is an obvious candidate for what changed in 2020.14 that might be causing the issue in `xija_gui_fit`. The 2021.1 release does not contain any package updates that are likely to cause this problem.

**Comment by jzuhone** [comment]
@taldcroft -- what I mean is the big python 3.8 update, and I agree with you that it's probably either the matplotlib (or maybe Qt/pyqt) update that caused the change in behavior.

**Comment by jzuhone** [comment]
Resolved by #108
